controllers/actions.github.com/ephemeralrunner_controller_test.go

```diff
@@ -477,7 +476,7 @@ var _ = Describe("EphemeralRunner", func() {
 					})
 					err = k8sClient.Status().Update(ctx, pod)
 					Expect(err).To(BeNil(), "Failed to update pod status")
-					return false, fmt.Errorf("pod haven't failed for 5 times.")
+					return false, nil
```

Comment:
why do we need to change this?

Reply:
I didn't find anything indicating that `return error` will make `Eventually` stop immediately. 😕

Reply:
For posterity, this passes:

```golang
func Test(t *testing.T) {
	g := gomega.NewWithT(t)
	i := 0
	g.Eventually(func() (int, error) {
		if i == 0 {
			i++
			return 0, errors.New("some error")
		}
		return i, nil
	}).Should(gomega.Equal(1))
}
```

🤦 , reverted.